### PR TITLE
Implement Redis v6 auth (fixes #182) 

### LINKED ISSUE #182: Please support redis 6 AUTH with user / password
I installed redis 6.0.11 on my ubuntu server via http://ppa.launchpad.net/redislabs/redis/ubuntu

I enabled /etc/redis/users.acl, added a password to user "default", and defined 2 more users with password:

```
user default allcommands allkeys on >secret
user alice allcommands allkeys on >ALICE2
user bob  on >bob3 ~bob:* +get
```

This works perfectly - I can swap users with redis-cli, GET and SET data:

```
127.0.0.1:6379> AUTH secret
OK
127.0.0.1:6379> SET bob:password xyz
OK
127.0.0.1:6379> AUTH alice ALICE2
OK
127.0.0.1:6379> SET bob:password xyz123
OK
127.0.0.1:6379> AUTH bob bob3
OK
127.0.0.1:6379> GET bob:password
"xyz123"
127.0.0.1:6379> SET bob:password supersecret
(error) NOPERM this user has no permissions to run the 'set' command or its subcommand
```

I got and compiled Webdis 0.1.13-dev and tried to define redis_auth in webdis.json:

* ```"secret"``` is OK 
* ```"alice ALICE2"``` receives ```{"GET":[false,"NOAUTH Authentication required."]}```

I tried to hack pool.c to have it using the full "user password" string from redis_auth, but didn't succeed. I also tested using ```redisAsyncCommand(ac, NULL, NULL, "AUTH alice ALICE2");``` instead of the current ```redisAsyncCommand(ac, NULL, NULL, "AUTH %s", p->cfg->redis_auth);``` and it works well.

Can you please implement Redis 6 AUTH command as defined on https://redis.io/commands/auth?

### PR DESCRIPTION
**Pull request by jessie-murray**
There are 3 commits in this PR:
1. 4640ac0 adds support for Redis 6 auth with a username and password
2. f159c7f fixes an issue I noticed in `slog.c` while writing this
3. d3d40bd adds logging for the `AUTH` command; webdis is currently ignoring the response from Redis and just relying on later commands failing to notify users that the current config doesn't work correctly

CC @nicolasff

### CODE REVIEW

**Review by nicolasff — CHANGES_REQUESTED** [review]
This looks great! Thanks so much for the PR. I think there is just one issue with the way an error is logged when auth is disabled, but otherwise all my other comments were minor suggestions.

File: src/conf.c [hunk 1]
```diff
@@ -294,3 +308,47 @@ conf_free(struct conf *conf) {
...
+
+static struct auth *
+conf_auth_username_password(json_t *jarray) {
+	size_t array_size = json_array_size(jarray);
+	if(array_size != 2) {
+		fprintf(stderr, ACL_ERROR_PREFIX "expected two values, found %zu" ACL_ERROR_SUFFIX, array_size);
+		return NULL;
+	}
+	json_t *jusername = json_array_get(jarray, 0);
+	json_t *jpassword = json_array_get(jarray, 1);
```
[the remark this answers is not in the document]
Comment (jessie-murray) [reply src/conf.c:333]: I see what you mean but this is not actually the case today, for example in this file itself there are variables declared inside of `if` blocks, like

```c
const char *plain = conf_string_or_envvar(json_string_value(jbasic));
```

File: src/conf.c [hunk 2]
```diff
@@ -294,3 +308,47 @@ conf_free(struct conf *conf) {
...
+	if(json_typeof(jusername) != JSON_STRING || json_typeof(jpassword) != JSON_STRING) {
+		fprintf(stderr, ACL_ERROR_PREFIX "both values need to be strings" ACL_ERROR_SUFFIX);
+		return NULL;
+	}
+
+	char *username = conf_string_or_envvar(json_string_value(jusername));
+	char *password = conf_string_or_envvar(json_string_value(jpassword));
+	struct auth *ret = calloc(1, sizeof(struct auth));
```
[the remark this answers is not in the document]
Comment (jessie-murray) [reply src/conf.c:341]: See response above.

File: src/conf.c [hunk 3]
```diff
@@ -16,6 +16,14 @@
 static struct acl *
 conf_parse_acls(json_t *jtab);
 
+#define ACL_ERROR_PREFIX "Config error with 'redis_auth': "
+#define ACL_ERROR_SUFFIX ". Starting with auth disabled.\n"
+
+static struct auth *
+conf_auth_legacy(char *password);
+static struct auth *
+conf_auth_username_password(json_t *jarray);
```
[the remark this answers is not in the document]
Comment (jessie-murray) [reply src/conf.c:25]: I started adding some but this adds a lot of noise to the code, I'm not sure how to say it. I find it a lot less readable and it's not super important in a file like this one. Maybe it makes more sense in other places where the code is not executed just one time?

**Comment by jessie-murray** [comment]
@nicolasff I added a new commit with the bugfix for the error message logged when auth is disabled, also I updated `conf_free` to fully free the `redis_auth` struct but then I noticed that it is never called since Webdis just exits with `exit(0)` so I was not really sure what to do.
